Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardEditViewModel.swift

```diff
@@ -51,26 +59,46 @@ class CreditCardEditViewModel: ObservableObject {
         return RemoveCardButton.AlertDetails(
             alertTitle: Text(CreditCardText.RemoveCardTitle),
             alertBody: nil,
-            primaryButtonStyleAndText: .destructive(Text(CreditCardText.RemovedCardLabel)),
+            primaryButtonStyleAndText: .destructive(Text(CreditCardText.RemovedCardLabel)) { [self] in
+                guard let creditCard = creditCard else { return }
+
+                removeSelectedCreditCard(creditCard: creditCard)
+            },
             secondaryButtonStyleAndText: .cancel(),
             primaryButtonAction: {},
             secondaryButtonAction: {}
         )
     }
 
-    init(profile: Profile) {
+    init(profile: Profile,
+         creditCard: CreditCard? = nil
+    ) {
         self.profile = profile
+        self.autofill = profile.autofill
+        self.creditCard = creditCard
     }
 
     init(profile: Profile = AppContainer.shared.resolve(),
          firstName: String,
          lastName: String,
          errorState: String,
-         enteredValue: String) {
+         enteredValue: String,
+         creditCard: CreditCard? = nil
+    ) {
         self.profile = profile
         self.firstName = firstName
         self.lastName = lastName
         self.errorState = errorState
         self.enteredValue = enteredValue
+        self.autofill = profile.autofill
+        self.creditCard = creditCard
+    }
+
+    // MARK: - Private helpers
+
+    private func removeSelectedCreditCard(creditCard: CreditCard) {
+        autofill.deleteCreditCard(id: creditCard.guid) { _, error in
+            // no-op
```

Comment:
Currently, no-op but later tickets in the epic will change this (showing a toast to confirm deletion, for example).